normandy/recipes/api/views.py

```diff
@@ -155,6 +144,101 @@ def disable(self, request, pk=None):
         recipe.disable()
         return Response(status=status.HTTP_204_NO_CONTENT)
 
+    @detail_route(methods=['GET'])
+    def approval_requests(self, request, pk=None):
+        recipe = self.get_object()
+        serializer = ApprovalRequestSerializer(recipe.approval_requests, many=True)
+        return Response(serializer.data)
+
+
+class ApprovalRequestViewSet(viewsets.ModelViewSet):
+    queryset = ApprovalRequest.objects.all()
+    serializer_class = ApprovalRequestSerializer
+    filter_fields = ('active',)
+    permission_classes = [
+        permissions.DjangoModelPermissionsOrAnonReadOnly,
+        AdminEnabledOrReadOnly,
+    ]
+
+    @transaction.atomic()
+    @reversion.create_revision()
+    def create(self, request, *args, **kwargs):
+        try:
+            return super().create(request, *args, **kwargs)
+        except IntegrityError:
+            return Response(
+                {'active': 'You can only have one open approval request for a recipe.'},
+                status=status.HTTP_400_BAD_REQUEST)
+
+    @transaction.atomic()
+    @reversion.create_revision()
+    def update(self, request, *args, **kwargs):
+        """
+        Intercept PUT requests and have them create instead of update
```

Comment:
I feel like we tried making a mixin or something for this and failed, but we really should figure out some way to avoid duplicating the code for this.